src/PerfCounters.cc

```diff
@@ -556,7 +578,9 @@ static void init_attributes() {
   // same thing.  Unclear if necessary.
   hw_interrupts_attr.exclude_hv = 1;
 
-  check_for_bugs();
+  if (ticks_for_branch == 0) {
```

Comment:
Overloading `ticks_for_branch` this way isn't great.

I think this might be a good time to switch the two bools and `ticks_for_branch` in `PmuConfig` to a flags word with flags, say, `FLAG_UNSUPPORTED`, `FLAG_BENEFITS_FROM_USELESS_COUNTER`, `FLAG_CHECK_FOR_INTEL_BUGS`, and `FLAG_UNCONDITIONAL_INDIRECT_JUMP_ADDS_TICK`.

Reply:
Or since this is local to `PerfCounters.cc` we can probably skip the `FLAG_` prefix.

Reply:
> Overloading `ticks_for_branch` this way isn't great.

You're right. It's appropriate for check_working_counter, I guess, but not for the others.

> I think this might be a good time to switch the two bools and `ticks_for_branch` in `PmuConfig` to a flags word with flags, say, `FLAG_UNSUPPORTED`, `FLAG_BENEFITS_FROM_USELESS_COUNTER`, `FLAG_CHECK_FOR_INTEL_BUGS`, and `FLAG_UNCONDITIONAL_INDIRECT_JUMP_ADDS_TICK`.

I don't like the idea of flag words much (as opposed to structs of booleans), but okay.

Reply:
The advantage of a flags word is that you get a default value of 0 for flags you don't explicitly set.